arbnode/seq_coordinator.go

```diff
@@ -492,6 +511,56 @@ func (c *SeqCoordinator) updateWithLockout(ctx context.Context, nextChosen strin
 	return c.noRedisError()
 }
 
+func (c *SeqCoordinator) deleteFinalizedMsgsFromRedis(ctx context.Context, finalized arbutil.MessageIndex) error {
+	updateFinalizedMsgCount := func() error {
+		finalizedBytes, err := c.msgCountToSignedBytes(finalized)
+		if err != nil {
+			return err
+		}
+		if err = c.Client.Set(ctx, redisutil.FINALIZED_MSG_COUNT_KEY, finalizedBytes, c.config.SeqNumDuration).Err(); err != nil {
+			return fmt.Errorf("couldn't set %s key to current finalizedMsgCount in redis: %w", redisutil.FINALIZED_MSG_COUNT_KEY, err)
+		}
+		return nil
+	}
+	prevFinalized, err := c.getRemoteFinalizedMsgCount(ctx)
+	if errors.Is(err, redis.Nil) {
+		var keys []string
+		for msg := finalized - 1; msg > 0; msg-- {
+			exists, err := c.Client.Exists(ctx, redisutil.MessageKeyFor(msg), redisutil.MessageSigKeyFor(msg)).Result()
+			if exists == 0 || err != nil {
```

Comment:
I had considered this scenario and that this case would happen at initialization as FINALIZED_MSG_COUNT_KEY wouldn't yet exist, and that even if this case does happen, the messages would just expire on their own after 1 day anyway. But I had misunderstood `Exists` error handling and thought that it returned redis.Nil in case of unknown key, my bad, its better to abort deleting and return an error here.

Yep we dont attempt to delete all keys until finalized-1 because that number would be too big.